### PR TITLE
Mobroute, Mobsql, Transito: init at 0.5.0

### PR DESCRIPTION
**Pull request by trungtruong1**
## Description of changes

This PR adds 3 packages: Mobroute, Mobsql, and Transito. 

[Mobroute](https://git.sr.ht/~mil/mobroute) is a general purpose FOSS public transportation router (e.g. trip planner) Go library and CLI that works by directly ingesting timetable (GTFS) data from transit agencies themselves (sourced from the [Mobility Database](https://database.mobilitydata.org/))

[Mobsql](https://git.sr.ht/~mil/mobsql): is a GTFS-to-SQL ETL tool which pulls GTFS sources from the MobilityDB into a SQLite database. It exposes both a CLI & a Go library (consumed by Mobroute).

[Transito](https://git.sr.ht/~mil/transito): is a simple Android & Linux GUI app to use Mobroute. It allows querying from/to via Nominatim and performing routing calculations on-the-go via Mobroute's underlying routing mechanism (implicitly integrates Mobsql as well).

This is my first contribution so please let me know if any changes or improvements are needed!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by trungtruong1** [comment]
Hi everyone,

I'm really sorry for the unexpected tagging/requesting reviews from "code owners" of other parts of the codebase that occurred. I'm still learning how to use git rebase and made a mistake. As someone who wants to contribute to Nixpkgs, I didn't mean to cause any trouble. I sincerely apologize for any inconvenience this caused.

Thank you for your understanding.

File: pkgs/by-name/tr/transito/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  src = fetchFromSourcehut {
+    owner = "~mil";
+    repo = "transito";
+    rev = "v${version}";
+    hash = "sha256-EM7hnHZjchpRzjl+iCBeLqLcyZhIS2Tl9opDx/w7Dcw=";
+  };
+
+  proxyVendor = true;
+
+  vendorHash = "sha256-8GLtZxn8q0VAmgs8fwqLNriKiRr2VjgvfFx9QdR4x5I=";
```
Comment (FliegendeWurst) [inline pkgs/by-name/tr/transito/package.nix:25]:
```suggestion
  vendorHash = "sha256-19vz7hsndGWYXgl/8I0P1pe3Kq+2GTUJMygk5ZW1lMY=";
```

File: pkgs/by-name/mo/mobroute/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{ lib
+, buildGoModule
```
Comment (FliegendeWurst) [inline pkgs/by-name/mo/mobroute/package.nix:2]: Please format with the new formatter.

File: pkgs/by-name/mo/mobroute/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    hash = "sha256-2NDZbv3jDM9INl/ZteSL6ege1vRhaeO5LYGBEMuSrmg=";
+  };
+
+  proxyVendor = true;
+
+  vendorHash = "sha256-meOYwLTkBAq7G0r9H5GtaT78Ztlxgo3Sfis/qwdC2dQ=";
+
+  buildPhase = ''
+    runHook preBuild
+    go build -o $GOPATH/bin/${pname} -tags=sqlite_math_functions -buildvcs=false cli/mobroutecli.go
```
Comment (FliegendeWurst) [inline pkgs/by-name/mo/mobroute/package.nix:23]: Do not use pname

#277994

File: pkgs/by-name/mo/mobroute/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    runHook postBuild
+  '';
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+    go test -failfast -p 1 -v -tags=sqlite_math_functions -buildvcs=false ./...
```
Comment (FliegendeWurst) [inline pkgs/by-name/mo/mobroute/package.nix:33]:
```suggestion
    go test -failfast -p 1 -v -tags=sqlite_math_functions -buildvcs=false ./...
    runHook postCheck
```

File: pkgs/by-name/mo/mobsql/package.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    runHook postBuild
+  '';
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+    go test -v ./...
```
Comment (FliegendeWurst) [inline pkgs/by-name/mo/mobsql/package.nix:33]:
```suggestion
    go test -v ./...
    runHook postCheck
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by McSinyx** [comment]
@FliegendeWurst, please direct your attention to GH-335613.

**Comment by FliegendeWurst** [comment]
Closing in favor of the linked PR

**Pull request closed without merging**